msgpack_class.c

```diff
@@ -528,16 +455,18 @@ static ZEND_METHOD(msgpack_unpacker, execute)
 
     if (unpacker->retval == NULL)
     {
-        ALLOC_INIT_ZVAL(unpacker->retval);
+        zval zv;
+        unpacker->retval = &zv;
```

Comment:
reference to a local stack value is undefined and dangerous behavior, you need move zval zv declaration out of this block.